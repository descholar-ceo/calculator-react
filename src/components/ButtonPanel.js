import {
  groupOne,
  groupTwo,
  groupThree,
  groupFour,
  groupFive,
  mapMarksToPanel,
} from '../helpers/mark-helpers';

const ButtonPanel = () => {
  const rowOne = mapMarksToPanel(groupOne());
  const rowTwo = mapMarksToPanel(groupTwo());
  const rowThree = mapMarksToPanel(groupThree());
  const rowFour = mapMarksToPanel(groupFour());
  const rowFive = mapMarksToPanel(groupFive());

  return (
    <div className="calc">
      <div className="btn-show">
        {rowOne}
      </div>

      <div className="btn-show">
        {rowTwo}
      </div>

      <div className="btn-show">
        {rowThree}
      </div>

      <div className="btn-show">
        {rowFour}
      </div>

      <div className="btn-show">
        {rowFive}
      </div>
    </div>
  );
};

export default ButtonPanel;
