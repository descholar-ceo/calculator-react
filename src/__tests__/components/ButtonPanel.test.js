import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../../components/ButtonPanel';

describe('Test ButtonPanel with snapshots', () => {
  it('Renders ButtonPanel correctly', done => {
    const buttonPanelComponent = renderer.create(<ButtonPanel clickHandler={() => 'anything'} />).toJSON();

    expect(buttonPanelComponent).toMatchSnapshot();
    done();
  });
});
