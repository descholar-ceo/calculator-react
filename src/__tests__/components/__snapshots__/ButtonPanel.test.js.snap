// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonPanel with snapshots Renders ButtonPanel correctly 1`] = `
<div
  className="button-panel-container"
>
  <div
    className="row-number-0"
  >
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange-bg"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row-number-1"
  >
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange-bg"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row-number-2"
  >
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange-bg"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row-number-3"
  >
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange-bg"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row-number-4"
  >
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="grey-bg"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange-bg"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
